turbopack/crates/turbopack-tests/tests/execution/turbopack/basic/comptime/input/index.js

```diff
@@ -172,3 +172,27 @@ it('should analyze numeric additions',()=>{
     require("fail");
   }
 })
+
+it('should consider side-effects of conditions', () => {
+  let sideEffects = 0;
+  function sideEffectFalse(value) {
+    sideEffects += value;
+    return false
+  }
```

Comment:
Wow, I didn't expect us to evaluate custom functions returning constant values...